pkg/extensions/search/cve/trivy/scanner.go

```diff
@@ -394,6 +394,13 @@ func (scanner Scanner) scanManifest(ctx context.Context, repo, digest string) (m
 				fixedVersion = "Not Specified"
 			}
 
+			var packagePath string
+			if vulnerability.PkgPath != "" {
+				packagePath = vulnerability.PkgPath
```

Comment:
We should get some coverage for this logic.

Maybe modify TestCVEStruct in pkg/extensions/search/cve/cve_test.go to have some packages provided by the mock CVE scanner and verify the packages are aggregated correctly.

Reply:
Actually you are correct, we don't have a way to mock this right now.
We'd have to test this with an actual vulnerable image.

Reply:
Interestingly enough,
this coverage report (https://app.codecov.io/gh/project-zot/zot/pull/2241/blob/pkg/extensions/search/cve/trivy/scanner.go#L390) indicates that the `FixedVersion` logic, which is similar to the package path is covered :)

Do we happen to have an actual image that is being pulled in during the tests?

Reply:
We do have such logic, look for the test with the comment below in pkg/extensions/search/cve/trivy/scanner_test.go

```
		// As of September 17 2023 there are 5 CVEs:
		// CVE-2023-1255, CVE-2023-2650, CVE-2023-2975, CVE-2023-3817, CVE-2023-3446
		// There may be more discovered in the future
```
There's a specific vulnerable layer downloaded into test/data/ by the makefile, and which has some known vulnerabilities.
It is easy to write a test to detect these, and not detect them, by simply not including that layer into the newer image.

PkgPath is more tricky to produce as not all CVEs packages have this property set.

Reply:
I tried to run that test file to see if I could bring in my own vulnerable layer, but I got this error when trying to run the existing test without any changes:
`zot/test/data/alpine/blobs/sha256/f56be85fc22e46face30e2c3de3f7fe7c15f8fd7c4e5add29d7f64b87abdaa09: no such file or directory'`

Is there something I need to download additionally in order for it to go through?

Reply:
You need to execute this make target https://github.com/project-zot/zot/blob/main/Makefile#L224

Reply:
Thank you!

I'm looking to add a new image or add a new layer to an existing zot test image.

The plan I had in mind was to copy a vulnerable JAR file into the image and use that image for testing or to get a fresh image (maybe alpine) and copy the file in.

May I ask how I could either update an existing zot test image or add a new image for zot test?

Reply:
I have some good news - I was able to get a test case that has a PackagePath verification to pass using a test image I built that has a vulnerable JAR inside.

It's a simple image based on Alpine. Is there a way I can commit the Dockerfile and have the image uploaded to the project-zot ghcr so I can use it in the test code?

Reply:
Sure, I can build and upload it manually for you. We don't have automation around that.

Reply:
Thanks!

Here are the details:
Directory:
```
.
├── Dockerfile
└── spring-web-5.3.31.jar

0 directories, 2 files
```
Spring JAR can be downloaded from here: wget https://repo1.maven.org/maven2/org/springframework/spring-web/5.3.31/spring-web-5.3.31.jar
Dockerfile contents:
```
FROM alpine:latest

RUN mkdir /usr/local/artifacts/
COPY spring-web-5.3.31.jar /usr/local/artifacts/spring-web-5.3.31.jar
```

Reply:
The image is at ghcr.io/project-zot/test-images/spring-web:5.3.31.
The jar file is in path /spring-web-5.3.31.jar.

I will open another PR with the changes. I also have another image which did not have any build instructions committed, and will include both.

Reply:
Thanks! Shall I wait for your PR with the changes to the Makefile before I update the commit with my new test case?

Reply:
The image is already there for you to use. The PR is just to commit the build instructions (https://github.com/project-zot/zot/pull/2249).

Reply:
Great! Sounds good.

Reply:
Made the changes to include coverage for these lines. Thanks again for the help with the image!